src/providers/arcgisrest/qgsafsshareddata.cpp

```diff
@@ -144,39 +144,62 @@ bool QgsAfsSharedData::getFeature( QgsFeatureId id, QgsFeature &f, const QgsRect
     return filterRect.isNull() || ( f.hasGeometry() && f.geometry().intersects( filterRect ) );
   }
 
-  // Fetch 100 features at the time
-  const int startId = ( id / 100 ) * 100;
-  const int stopId = std::min< size_t >( startId + 100, mObjectIds.length() );
+  bool featureFetched = false;
+  int startId, stopId;
   QList<quint32> objectIds;
-  objectIds.reserve( stopId );
-  for ( int i = startId; i < stopId; ++i )
-  {
-    if ( i >= 0 && i < mObjectIds.count() && !mDeletedFeatureIds.contains( i ) && !mCache.contains( i ) )
-      objectIds.append( mObjectIds.at( i ) );
-  }
+  QVariantMap queryData;
+  while ( !featureFetched )
+  {
+    startId = ( id / mMaximumFetchObjectsCount ) * mMaximumFetchObjectsCount;
+    stopId = std::min< size_t >( startId + mMaximumFetchObjectsCount, mObjectIds.length() );
+    objectIds.clear();
+    objectIds.reserve( stopId );
```

Comment:
This should be

```suggestion
    objectIds.reserve( stopId - startId );
```

shouldn't it?

Reply:
Correct.